### PR TITLE
Reloading fixes [RDY]

### LINKED ISSUE #15315: Reload time depends on total ammo volume rather than stack size volume
With 50 ammo, reloading one round in a Ruger Redhawk takes 6 seconds on a character with 100 speed. With 1000 ammo, it takes 72 seconds. The time should logically depend on the size of each individual cartridge, rather than the total stack volume, and inversely on the amount of ammo (finding one cartridge in a mountain of them should be easy). Even if the handling cost disregards the ease of finding the item, you're not handling 1000 cartridges, you're handling one at a time.

### LINKED ISSUE #16373: Number of turns to reload guns don't change when changing amount of ammo to be reloaded
Seems like to happen only with revolvers.

Win 10, tiles, 0.C-4750.

### PR DESCRIPTION
**Pull request by mugling**
Factored out of a larger PR that I'm working

Fixes #16373
Fixes #15315

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -10008,11 +10008,19 @@ void player::mend_item( item_location&& obj, bool interactive )
     }
 }
 
-int player::item_handling_cost( const item& it, bool effects, int factor ) const {
-    int mv = std::max( 1, it.volume() * factor );
+int player::item_handling_cost( const item& it, bool effects, int factor, int qty ) const {
+
+    // If necessary crete duplicate with appropriate number of charges
```
Comment (illi-kun) [inline src/player.cpp]: "crete" - that's a typo, I suppose.

**Comment by Coolthulhu** [comment]
Bugs out when reloading a flamethrower form bottles. Will debugmsg that it isn't a magazine or ammo, then load only one charge.

**Comment by mugling** [comment]
@Coolthulhu well spotted. ad66b6f includes a fix for items using `gasoline` as ammo and also makes some effort towards making their reload times more consistent (I'd prefer to do balance in a separate PR given this is a bugfix)

File: data/json/items/gun/gasoline.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    "material": ["superalloy", "ceramic"],
+    "range": 12,
+    "ranged_damage": 10,
+    "dispersion": 240,
+    "sight_dispersion": 150,
+    "durability": 9,
+    "burst": 5,
+    "clip_size": 2000,
+    "valid_mod_locations": [ [ "accessories", 4 ], [ "rail", 1 ], [ "grip", 1 ], [ "stock", 1 ], [ "underbarrel", 1 ] ],
+    "extend": { "flags": [ "MODE_BURST" ] }
```
Comment (Rivet-the-Zombie) [inline data/json/items/gun/gasoline.json]: The RM451 is supposed to use ````"FIRE_20"````.

Comment (mugling) [inline data/json/items/gun/gasoline.json]: > The RM451 is supposed to use "FIRE_20".

Fixed in 3262d1f

File: data/json/items/gun/gasoline.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
+[
+  {
+    "id": "flamethrower",
+    "copy-from": "flamethrower_base",
+    "type": "GUN",
```
Comment (kevingranade) [inline data/json/items/gun/gasoline.json]: Isn't type redundant with the base?

File: src/player.cpp [hunk 4]
```diff
@@ -10008,11 +10008,19 @@ void player::mend_item( item_location&& obj, bool interactive )
     }
 }
 
-int player::item_handling_cost( const item& it, bool effects, int factor ) const {
-    int mv = std::max( 1, it.volume() * factor );
+int player::item_handling_cost( const item& it, bool effects, int factor, int qty ) const {
+
+    // If necessary create duplicate with appropriate number of charges
```
Comment (kevingranade) [inline src/player.cpp]: I don't know how often this comes up, if it's more than two times or so it's be good to wrap it in a method.

File: data/json/items/gun/gasoline.json (hunk 3 above)
Comment (mugling) [inline data/json/items/gun/gasoline.json]: Yes, let me me refactor that some

File: src/player.cpp (hunk 4 above)
Comment (mugling) [inline src/player.cpp]: Possibly, but I want to rework handling costs further so I'd prefer to refactor after this

**Comment by mugling** [comment]
Did part of this get pushed?

**Pull request closed without merging**